File: app/src/main/java/bassamalim/hidaya/database/AthkarDao.java
package bassamalim.hidaya.database;

import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

@Dao
public interface AthkarDao {

    @Query("SELECT * FROM athkar")
    List<AthkarDB> getAll();

    @Query("SELECT * FROM athkar WHERE category_id = :category")
    List<AthkarDB> getList(int category);

    @Query("SELECT athkar_name FROM athkar WHERE category_id = :category AND athkar_id = :id")
    String getName(int category, int id);

/*
    @Query("SELECT * FROM hafs_ayat WHERE id IN (:userIds)")
    List<JAyah> loadAllByIds(int[] userIds);

    @Query("SELECT * FROM user WHERE first_name LIKE :first AND " +
            "last_name LIKE :last LIMIT 1")
    JAyah findByName(String first, String last);

    @Insert
    void insertAll(JAyah... users);

    @Delete
    void delete(JAyah user);
    */
}

File: app/src/main/java/bassamalim/hidaya/core/data/database/daos/AthkarDao.kt
package bassamalim.hidaya.core.data.database.daos

import androidx.room.Dao
import androidx.room.Query
import bassamalim.hidaya.core.data.database.dbs.AthkarDB

@Dao
interface AthkarDao {
    @Query("SELECT * FROM athkar")
    fun getAll(): List<AthkarDB>

    @Query("SELECT athkar_name FROM athkar")
    fun getNames(): List<String>

    @Query("SELECT athkar_name_en FROM athkar")
    fun getNamesEn(): List<String>

    @Query("SELECT * FROM athkar WHERE favorite = 1")
    fun getFavorites(): List<AthkarDB>

    @Query("SELECT * FROM athkar WHERE category_id = :category")
    fun getList(category: Int): List<AthkarDB>

    @Query("SELECT athkar_name FROM athkar WHERE athkar_id = :id")
    fun getName(id: Int): String

    @Query("SELECT athkar_name_en FROM athkar WHERE athkar_id = :id")
    fun getNameEn(id: Int): String

    @Query("UPDATE athkar SET favorite = :value WHERE athkar_id = :id")
    fun setFav(id: Int, value: Int)

    @Query("SELECT favorite FROM athkar")
    fun getFavs(): List<Int>
}